src/main/java/space/obminyashka/items_exchange/controller/AuthController.java

```diff
@@ -128,4 +145,25 @@ public void logout(HttpServletRequest req) {
 
         throw new BadRequestException(getMessageSource("user.not-registered"));
     }
+
+    @PostMapping(value = "/refresh/token")
+    @ApiOperation(value = "Renew access token with refresh token")
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "OK"),
+            @ApiResponse(code = 401, message = "UNAUTHORIZED"),
+            @ApiResponse(code = 404, message = "NOT FOUND")
+    })
+    @ResponseStatus(HttpStatus.OK)
+    public RefreshTokenResponse refreshToken(@Valid @RequestBody RefreshTokenRequest request) throws RefreshTokenException {
+        final var token = request.getRefreshToken();
+        final var refreshToken = refreshTokenService.findByToken(token)
+                .orElseThrow(() -> new EntityNotFoundException(getParametrizedMessageSource("refresh.token.not-found", token)));
```

Comment:
Let's make it better....
Right now the purpose that method it to find token into DB, to check if it still alive, and to recreate the token.
Having Optional here allows doing that much better in functional style:
```java

final var accessToken = refreshTokenService.findByToken(token)
                .filter(Predicate.not(refreshTokenService::isRefreshTokenExpired))
                .map(RefreshToken::getUser)
                .map(user -> jwtTokenProvider.createToken(user.getUsername(), user.getRole()))
                .orElseThrow(() -> new RefreshTokenException("some great exception message");
        
        return new RefreshTokenResponse(accessToken, ...
```
What do you think about it (the naming is up to you)?

Reply:
Additional thought: as for me the logic above can be moved into a service's new method.

Reply:
Ok, thank you!